modules/internal/ChapelArray.chpl

```diff
@@ -1379,16 +1452,17 @@ module ChapelArray {
         ranges(i) = dim(i).chpl__unTranslate(off(i));
       var d = _value.dsiBuildRectangularDom(rank, _value.idxType,
                                            _value.stridable, ranges);
-      if !noRefCount then
-        if (d.linksDistribution()) then
-          d.dist.incRefCount();
+      // Since we've created a new domain, the distribution needs to
+      // live at least as long as this new domain.
+      if d.linksDistribution() then
+        d.dist.add_dom(d);
```

Comment:
yes, but I worry that would make it less clear what is going on. It would only move us from 2 lines to 1 line, so I'm not sure if it's worth it in this case.